pkg/reaper/schema.go

```diff
@@ -0,0 +1,51 @@
+package reaper
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/reaper/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+)
+
+func ReconcileBackendSchema(
```

Comment:
So I thought about that, but I could see a potential problem:

The extracted function is likely to have this signature:

```
func EnsureReplication(keyspaceName, clusterName string, replication map[string]int , managementApi cassandra.ManagementApiFacade, logger logr.Logger) error
```

As you can see it would reference the `cassandra` package. If this function is placed in the `utils` package, then there could be no reference to `utils` from `cassandra` otherwise we would face a circular reference.

Note that currently no such reference exists; but I believe this could change in the future.

If we want to go forward with the common function approach, I would recommend moving the `management.go` file to its own package, then placing the `EnsureReplication` function in that package.

Wdyt?

Reply:
In fact, I realize that we could turn `EnsureReplication` into a method of the `ManagementApiFacade` interface. I think this is the most elegant solution. I will go with that for now but let me know if you agree.

Reply:
I also moved the existing, replication-related stuff from `utils` to `cassandra`. It looks better this way imo.

Reply:
Sorry for the delayed response. Both changes sound good.